subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c

```diff
@@ -1266,8 +1267,13 @@ int lwm2m_init_firmware_cb(lwm2m_firmware_event_cb_t cb)
 		"application", firmware_block_received_cb, firmware_update_cb);
 	lwm2m_firmware_object_setup_init(application_obj_id);
 
-	modem_obj_id = lwm2m_adv_firmware_create_inst(
-		"modem:" CONFIG_SOC, firmware_block_received_cb, firmware_update_cb);
+	static char hw_buf[sizeof("modem:nRF91__ ____ ___ ")];
+
+	strcat(hw_buf, "modem:");
+	modem_info_get_hw_version(hw_buf + strlen("modem:"), strlen(hw_buf) - strlen("modem:"));
```

Comment:
modem_info_get_hw_version seems to want the size of the target buffer. As this is using strlen to calculate, it will ignore the null character and send the remaining buffer space without it?

Reply:
Good catch.
Looks like I made an error on last refactoring. It should have been `sizeof() - strlen()`